### PR TITLE
Upgrade Rubocop

### PR DESCRIPTION
**Pull request by rzane**
### Summary

> Provide a general description of the code changes in your pull
request... were there any bugs you had fixed? If so, mention them. If
these bugs have open GitHub issues, be sure to tag them here as well,
to keep the conversation linked together.

The specified Rubocop version was quite old. Running the specs against Ruby 3.1 isn't possible without upgrading due to [this issue](https://github.com/rubocop/rubocop/issues/10258).

### Other Information

> If there's anything else that's important and relevant to your pull
request, mention that information here. This could include
benchmarks, or other information.

> Thanks for contributing to TestTrack!

/domain @Betterment/test_track_core
/platform @Betterment/test_track_core

### CODE REVIEW

File: test_track_rails_client.gemspec [hunk 1]
```diff
@@ -31,7 +31,7 @@ Gem::Specification.new do |s|
   s.add_development_dependency 'pry-rails'
   s.add_development_dependency 'rails-controller-testing'
   s.add_development_dependency 'rspec-rails'
-  s.add_development_dependency 'rubocop', '~> 0.81.0'
+  s.add_development_dependency 'rubocop', '~> 1.48.0'
   s.add_development_dependency 'rubocop-performance'
   s.add_development_dependency 'rubocop-rails'
```
Comment (smudge) [inline test_track_rails_client.gemspec:36]: Thought son switching these all over to `betterlint`?

Comment (rzane) [reply test_track_rails_client.gemspec:36]: I'm not opposed, but I'd rather not prioritize that right now.

**Review by smudge — APPROVED** [review]
domain LGTM

**Review by smudge — APPROVED** [review]
platform LGTM as well

**Pull request merged**